### PR TITLE
Reminders

### LINKED ISSUE #86: Reminder command

### PR DESCRIPTION
**Pull request by pedrofracassi**
Closes #86

- [x] **Database schema**
  - Reminder id
    - Maybe 5-letter codes, just like URL shorteners?
  - User id
  - Message id
  - Channel id
  - Text (optional)
  - Timestamp
- [ ] **`s!remindme` command**
  - Should parse human-readable date/time strings, just like the reddit RemindMe bot. [This can probably be done with Datejs](https://github.com/datejs/Datejs), but I was thinking of writing our own library based on it. **See [bladiem](https://github.com/SwitchbladeBot/bladiem).**
  - Should allow users to set a text for the reminder with `"double-quoted strings"` or a `--text` flag
  - `s!reminder cancel <id>` should be a thing
- [ ] **Trigger the reminders at the right time**
  - This would remove the reminder from the database, obviously.

### CODE REVIEW

**Comment by metehus** [comment]
#771

**Comment by Flame-0** [comment]
image
image
image
image

**Comment by pedrofracassi** [comment]
Where

File: src/database/mongo/schemas/ReminderSchema.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { Schema } = require('mongoose')
+
+module.exports = new Schema({
+  _id: String,
```
Comment (fernando457829) [inline src/database/mongo/schemas/ReminderSchema.js]: _id prop is unnecessary, default in any mongoose schema

**Comment by fernando457829** [comment]
why other repository? User.reminders is bettter

UserSchema:
```js
{
  // ...currentProps
  reminders: [{ type: Schema.Types.ObjectId, ref: 'Reminder' }] // using populate to fill. read
}```

To fill reminders use populate method, read more in https://mongoosejs.com/docs/populate.html

ReminderSchema:
```js
{
  user: [{ type: Schema.Types.ObjectId, ref: 'User' }],
  messageId: String,
  channelId: String,
  text: String,
  timestamp: String
}
```

File: src/commands/utility/reminder.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const { Command, SwitchbladeEmbed } = require('../../')
+
+module.exports = class Reminder extends Command {
+  constructor (client) {
+    super(client, {
+      name: 'reminder',
+      aliases: ['remindme'],
+      category: 'utility',
+      parameters: [{
+        type: 'string',
```
Comment (fernando457829) [inline src/commands/utility/reminder.js]: DateParameter?

File: src/database/mongo/schemas/ReminderSchema.js (hunk 1 above)
Comment (dpaiv0) [reply src/database/mongo/schemas/ReminderSchema.js]: it's useful, since _id equals the user's Discord ID

Comment (dpaiv0) [reply src/database/mongo/schemas/ReminderSchema.js]: oh, on reminders...

Comment (dpaiv0) [reply src/database/mongo/schemas/ReminderSchema.js]: nvm go ahead

File: src/commands/utility/reminder.js (hunk 2 above)
Comment (dpaiv0) [reply src/commands/utility/reminder.js]: cbd when bladiem becomes a thing HAhaa

**Comment by metehus** [comment]
reminder list?

**Pull request closed without merging**